torch/csrc/jit/JIT-AUTOCAST.md

```diff
@@ -169,6 +170,32 @@ def traced(a, b):
 torch.jit.trace(traced, (x, y))
 ```
 
+#### Mixing eager autocast and scripting
+
+Calling un-autocast scripted functions while using eager autocast has some
+limitations:
```

Comment:
This sounds a little vague to me. I thought after this sentence, we are going to discuss the limitations. but the example below looks legit. Let's make it clear that this is the "proper" way to mix the two?

Reply:
@jjsjann123 the example below **fails**. I think the issue is when it is combined with autodiff - the eager mode autocasting during forward causes the output of linear() to be fp16; but then in the backward pass grad_linear_input ends up being fp16 since autodiff doesn't know that there was a casting op.

btw, do you know how eager mode autocasting + autograd handles this?

```
Traceback (most recent call last):
  File "/fsx/users/dberard/scripts/logscript/autocast_bad.py", line 18, in <module>
    res.backward()
  File "/fsx/users/dberard/pytorch/torch/_tensor.py", line 399, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/fsx/users/dberard/pytorch/torch/autograd/__init__.py", line 173, in backward
    Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
  File "<string>", line 37, in <backward op>
            def backward(grad_output):
                output_mask = [True, weight is not None, bias is not None]
                grad_input, grad_weight, grad_bias = torch.native_layer_norm_backward(grad_output, input, normalized_shape, mean, rstd, weight, bias, output_mask)
                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
                return grad_input, None, grad_weight, grad_bias, None, None
            return output, backward
RuntimeError: expected scalar type Float but found Half
```

Reply:
This sounds strange... We do have the problem of implicit casting on ops messing up dtype for backwards. But I don't see how that's happening here.
We have the function jitted, so forward should run with JIT and no eager autocast should be in play here. `linear` is marked as fp16 op https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/autocast.cpp#L334, so we should see an explicit cast op in autodiff.

Maybe there's something else going wrong. Let me double check that